### PR TITLE
feat: remove XMLBeans dependency from databind module

### PR DESCRIPTION
**Pull request by david-waltermire**
## Summary

- Replace Apache XMLBeans with Metaschema-based binding classes for configuration parsing
- Implement pure Java name conversion utilities (replacing XMLBeans NameUtil)
- Add metaschema-bindings.yaml module defining the binding configuration schema
- Pre-generate binding classes using bootstrap pattern (avoiding circular dependency)
- Fix SARIF module binding configuration (pre-existing issue with wrong namespace)

### databind module
- Add `metaschema-bindings.yaml` - Metaschema module for binding configuration
- Add pre-generated binding classes in `config/binding/` package (bootstrap pattern)
- Add package-info.java documenting the bootstrap regeneration process
- Update `ClassUtils.java` - Pure Java implementation of name conversions
- Add Java reserved words and contextual keywords (module, var, yield, record, sealed, permits)
- Update `DefaultBindingConfiguration.java` - Use new Metaschema bindings
- Remove XMLBeans XSD schema and configuration files
- Remove XMLBeans plugin and dependency from pom.xml
- Add PMD/Checkstyle/JaCoCo exclusions for generated code

### databind-metaschema module
- Fix SARIF binding configuration namespace and package name
- Add `<configs>` reference to sarif-codegen execution

### Tests
- Add 40 unit tests for ClassUtils name conversion utilities
- Cover edge cases: digit prefixes, reserved words, contextual keywords, underscore-only parts

## Test plan

- [x] Build passes: `mvn clean install -PCI -Prelease`
- [x] All ClassUtils tests pass (40 tests)
- [x] Existing databind functionality unchanged
- [x] SARIF module builds correctly

## Related

PRD: `PRDs/20251221-xmlbeans-removal/`

## Release Notes

* **Documentation**
  * Added migration planning documentation for replacing XMLBeans with Metaschema bindings across databind and metaschema-testing modules.

* **Refactor**
  * Replaced XMLBeans with Metaschema-based binding framework for configuration parsing and code generation.
  * Updated module structure and package organization; removed XMLBeans dependencies and build artifacts.
  * Enhanced name transformation utilities with improved URI-to-package conversion logic.

* **Tests**
  * Added comprehensive unit tests for binding configuration and name transformation utilities.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: databind/src/main/java/gov/nist/secauto/metaschema/databind/codegen/ClassUtils.java [hunk 1]
```diff
@@ -62,20 +66,245 @@ public static String toVariableName(@NonNull String name) {
...
+   */
+  private static final Set<String> JAVA_RESERVED_WORDS = Set.of(
+      "abstract", "assert", "boolean", "break", "byte", "case", "catch", "char", "class", "const",
+      "continue", "default", "do", "double", "else", "enum", "extends", "final", "finally", "float",
+      "for", "goto", "if", "implements", "import", "instanceof", "int", "interface", "long", "native",
+      "new", "package", "private", "protected", "public", "return", "short", "static", "strictfp",
+      "super", "switch", "synchronized", "this", "throw", "throws", "transient", "try", "void",
+      "volatile", "while", "true", "false", "null",
+      // Contextual keywords (Java 9+)
+      "module", "var", "yield", "record", "sealed", "permits");
```
[the remark this answers is not in the document]
Comment (david-waltermire) [reply databind/src/main/java/gov/nist/secauto/metaschema/databind/codegen/ClassUtils.java:298]: The comment is ok as-is
